lib/package-detail-view.js

```diff
@@ -41,7 +41,10 @@ export default class PackageDetailView {
     const packageRepoClickHandler = (event) => {
       event.preventDefault()
       const repoUrl = this.packageManager.getRepositoryUrl(this.pack)
-      if (repoUrl) {
+      const repoName = url.parse(repoUrl).pathname
```

Comment:
> @rsese sure! Actually tried to find the test for the previous behaviour but couldn't find it. Do you have any example of similar tests? Maybe in another package?

It might also be useful to look at: https://github.com/atom/settings-view/pull/1087/commits/fec6b30c230c24b22cfa98042a02c8856306d85f since the repository link seems to have no click handler.

Reply:
> It looks like repoUrl might not be defined here; is it handled correctly?

I agree, @vinkla probably needs to change `const repoUrl` to `let repoUrl`.

Reply:
I agree with @Aerijo's assessment that this will trigger unhandled exceptions if `repoUrl` is undefined — actually if it's not a string. I suggest changing the proposed implementation to something like this

``` js
const repoUrl = this.packageManager.getRepositoryUrl(this.pack)
if (typeof repoUrl === 'string') {
  const repoPath = url.parse(repoUrl).pathname
  if (repoPath === '/atom/atom') {
    shell.openExternal(`${repoUrl}/tree/master/packages/${this.pack.name}`)
  } else {
    shell.openExternal(repoUrl)
  }
}
```

Alternatively the `repoPath` variable can be inlined since it's only used once

``` js
const repoUrl = this.packageManager.getRepositoryUrl(this.pack)
if (typeof repoUrl === 'string') {
  if (url.parse(repoUrl).pathname === '/atom/atom') {
    shell.openExternal(`${repoUrl}/tree/master/packages/${this.pack.name}`)
  } else {
    shell.openExternal(repoUrl)
  }
}
```

I personally prefer the first version because it's slightly more explicit and easier to read.

Reply:
Thanks for your suggestion guys! I've pushed an update. 

@maralv thanks for linking the tests. I'll take a look.